src/Symfony/Component/Console/CommandLoader/CommandLoaderInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Symfony\Component\Console\CommandLoader;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+interface CommandLoaderInterface
+{
+    /**
+     * Loads a command.
+     *
+     * @param string $name
+     *
+     * @return Command
+     *
+     * @throws CommandNotFoundException
+     */
+    public function get($name);
+
+    /**
+     * Checks if a command exists.
+     *
+     * @param string $name
+     *
+     * @return bool
+     */
+    public function has($name);
+
+    /**
+     * @return iterable All registered commands mapped by names
+     */
+    public function all();
```

Comment:
unless I'm misread, this method is not used at all, better removed it then, esp. since getNames() already allows iterating when needed